File: Tests/ButterflyNet.Tests/DefinitionListTests.cs
using NUnit.Framework;

namespace ButterflyNet.Parser.Tests {
	[TestFixture]
	public class DefinitionListTests : WikiToHtmlTest {
		[Test]
		public void Should_parse_definition_list() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn(";term\n:definition"), "<dl><dt>term</dt><dd>definition</dd></dl>");
		}

		[Test]
		public void Should_allow_formatting_in_definitions() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn(";__te''rm''__\n:definition"), "<dl><dt><strong>te<em>rm</em></strong></dt><dd>definition</dd></dl>");
		}

		[Test]
		public void Should_allow_multiple_paragraphs_inside_definition() {
			const string text = @";term
:{definition

that spans

multiple lines}:";

			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn(text), "<dl><dt>term</dt><dd><p>definition</p><p>that spans</p><p>multiple lines</p></dd></dl>");
		}

		[Test]
		public void Should_allow_formatting_in_multiline_definition() {
			const string text = @";term
:{__definition__}:";

			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn(text), "<dl><dt>term</dt><dd><p><strong>definition</strong></p></dd></dl>");
		}
	}
}

File: src/js/tests/DefinitionListTests.js
var definitionListTests = function() {
	var parser;
	
	return {
		setup: function() {
			parser = createParser();
		},
		
		tearDown: function() {
			parser = null;
		},
		
		test: function Definition_list_tests() {
			return [
				function Should_parse_definition_list() {
					Assert.that(trimLf(parser.parseAndReturn(";term\n:definition")), Is.equalTo("<dl><dt>term</dt><dd>definition</dd></dl>"));
				},
				
				function Should_allow_formatting_in_definitions() {
					Assert.that(trimLf(parser.parseAndReturn(";__te''rm''__\n:definition")), Is.equalTo("<dl><dt><strong>te<em>rm</em></strong></dt><dd>definition</dd></dl>"));
				},
				
				function Should_allow_multiple_paragraphs_inside_definition() {
					Assert.that(
						trimLf(parser.parseAndReturn(";term\n:{definition\n\nthat spans\n\nmultiple lines}:")), 
						Is.equalTo("<dl><dt>term</dt><dd><p>definition</p><p>that spans</p><p>multiple lines</p></dd></dl>")
					);
				},
				
				function Should_allow_formatting_in_multiline_definition() {
					Assert.that(trimLf(parser.parseAndReturn(";term\n:{__definition__}:")), Is.equalTo("<dl><dt>term</dt><dd><p><strong>definition</strong></p></dd></dl>"));
				},
				
				function Should_not_create_list_if_inline_scope_is_not_closed() {
					Assert.that(trimLf(parser.parseAndReturn("__foo\n;term__")), Is.equalTo("<p><strong>foo;term</strong></p>"));
				}
			];
		}
	};
}();

Jarvis.run(definitionListTests);